esp/esdlscriptlib/esdl_script.cpp

```diff
@@ -1119,14 +1123,56 @@ class CEsdlTransformOperationHttpHeader : public CEsdlTransformOperationWithoutC
 
     bool processHeader(IEsdlScriptContext * scriptContext, IXpathContext * targetContext, IXpathContext * sourceContext, IProperties *headers) override
     {
-        CXpathContextLocation location(targetContext);
-        targetContext->addElementToLocation("header");
-        StringBuffer name;
+        StringBuffer name, value;
         m_name.get(name, *sourceContext);
-
-        StringBuffer value;
         if (m_value)
             sourceContext->evaluateAsString(m_value, value);
+        if (m_secretSpec.name && streq(name, "Authorization"))
+            generateAuthorizationValue(scriptContext, sourceContext, value);
+        return processHeaderValue(scriptContext, targetContext, sourceContext, headers, name, value);
+    }
+
+    virtual void toDBGLog () override
+    {
+    #if defined(_DEBUG)
+        DBGLOG ("> %s (%s, value(%s)) >>>>>>>>>>", m_tagname.str(), m_name.configValue(), m_value ? m_value->getXpath() : "");
+    #endif
+    }
+
+protected:
+    void generateAuthorizationValue(IEsdlScriptContext * scriptContext, IXpathContext * sourceContext, StringBuffer& method)
+    {
+        HttpConnectSecretId id(m_secretSpec, sourceContext);
+        if (id.name.isEmpty())
+            recordException(ESDL_SCRIPT_InvalidOperationAttr, "empty or missing http-connect secret name");
+        Owned<IPTree> secret(scriptContext->getSecret("esp", id));
+        if (!secret)
+            recordException(ESDL_SCRIPT_MissingOperationAttr, "missing http-connect secret");
+
+        if (method.isEmpty() || streq(method, "Basic")) 
```

Comment:
You are correct about "Authorization" needing to be checked insensitively. "Basic" is part of the value. Values can be case sensitive, and I only see it presented one way.

Reply:
Just to make things more interesting, httpbinding.cpp does a case insensitive comparison of "Basic" while httplib.h performs a case sensitive comparison of "Basic". We appear to be inconsistent on handling of the value.

Fortunately, the question is only relevant because I try to reuse the value in the outgoing header instead of treating it as a simple selector. I'll update to treat the input value as a case-insensitive selector and replace it with the header value that seems to be expected.